File: tests/basic/statements/SwitchTest.java
package tests.basic.statements;

public class SwitchTest {

  public static void main(String[] args) {
    for (int i = 0; i < 3; i++) {
      switch (i) {
      case 0:
        System.out.println("Salut!");
        break;
      case 1:
        System.out.println("Foo");
        break;
      case 2:
        System.out.println("Bar");
        break;
      default:
        break;
      }
    }
    for (int i = 0; i < 4; i++) {
      switch (i) {
      case 0 -> System.out.println("John");
      case 1 -> System.out.println("Doe");
      case 2 -> System.out.println("420");
      default -> System.out.println("none");
      }
    }
    for (int i = 0; i < 4; i++) {
      String s = "s"+i;
      switch (s) {
      case "s0" -> System.out.println("Booz");
      case "s1" -> System.out.println("Baz");
      case "s2" -> System.out.println("Fuu");
      default -> System.out.println("none");
      }
    }
  }
}

File: tests/basic/statements/switch.test.ts

import ClassManager from "@base/ClassLoader"
import Program from "@base/Program"
import { createNewTestClasses } from "@base/Stubs"
import TestSystem from "@stub/tests/TestSystem"
import { expect, test } from "bun:test"

test("statement: switch", () => {
	const className = "tests/basic/statements/SwitchTest"
	const stubClasses = createNewTestClasses()

	const classManager = new ClassManager(stubClasses)
	const system = stubClasses.getStubClass("java.lang.System") as TestSystem
	const consoleOutput = system.out

	const main = classManager.get(className)
	main.executeMethod("main", classManager)

	// Program.debug = true
  expect(consoleOutput.printlnLines[0]).toBe("Salut!")
  expect(consoleOutput.printlnLines[1]).toBe("Foo")
  expect(consoleOutput.printlnLines[2]).toBe("Bar")
  expect(consoleOutput.printlnLines[3]).toBe("John")
  expect(consoleOutput.printlnLines[4]).toBe("Doe")
  expect(consoleOutput.printlnLines[5]).toBe("420")
  expect(consoleOutput.printlnLines[6]).toBe("none")
	expect(consoleOutput.printlnLines[7]).toBe("Booz")
	expect(consoleOutput.printlnLines[8]).toBe("Baz")
	expect(consoleOutput.printlnLines[9]).toBe("Fuu")
	expect(consoleOutput.printlnLines[10]).toBe("none")
	// Program.debug = false
})
